### PR TITLE
feat: Add run_xxx generate proto representation

### PR DESCRIPTION
**Pull request by typhoonzero**
Generate protobuf representations when writing `couler.run_xxx` APIs.

### CODE REVIEW

File: couler/core/utils.py [hunk 1]
```diff
@@ -274,3 +276,8 @@ def bool_to_str(bool_val):
     if not isinstance(bool_val, bool):
         raise TypeError("The bool_val is required to be boolean type")
     return "true" if bool_val else "false"
+
+
+def get_uniq_step_id():
+    global STEP_ID
```
Comment (terrytangyuan) [inline couler/core/utils.py:282]: Why is this needed? Could we move this to states.py?

File: couler/core/proto_repr.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from couler.core import utils
+from couler.proto import couler_pb2
+
+DEFAULT_WORKFLOW = None
```
Comment (terrytangyuan) [inline couler/core/proto_repr.py:17]: Move to states.py?

File: couler/core/run_templates.py [hunk 3]
```diff
@@ -189,9 +195,24 @@ def run_container(
...
 
     rets = _container_output(step_name, func_name, _output)
     states._steps_outputs[step_name] = rets
+
+    pb_step = proto_repr.step_repr(  # noqa: F841
+        step_name,
+        func_name,
+        image,
+        command,
+        None,
```
Comment (terrytangyuan) [inline couler/core/run_templates.py:210]: Add arg names so the code is more readable

**Review by terrytangyuan — CHANGES_REQUESTED** [review]

File: couler/core/utils.py (hunk 1 above)
Comment (typhoonzero) [reply couler/core/utils.py]: Moved to `proto_repr.py` since it's only used in that file.

File: couler/core/proto_repr.py (hunk 2 above)
Comment (typhoonzero) [reply couler/core/proto_repr.py:17]: It's used only in the current file, no need to put it some where else.

File: couler/tests/proto_repr_test.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import unittest
+
+import couler.argo as couler
+from couler.core.proto_repr import get_default_proto_workflow
+
+
+class ArtifactTest(unittest.TestCase):
```
Comment (terrytangyuan) [inline couler/tests/proto_repr_test.py:7]: Update the incorrect test class name `ArtifactTest`

**Review by terrytangyuan — APPROVED** [review]
